pkg/startup/v7/startup.go

```diff
@@ -101,48 +102,70 @@ func (s *startup) Hash(role api.AgentPoolProfileRole) ([]byte, error) {
 func (s *startup) writeFiles(role api.AgentPoolProfileRole, w writers.Writer, hostname, domainname string) error {
 	assetNames := AssetNames()
 	sort.Strings(assetNames)
+	var filesToWrite = map[string]string{}
```

Comment:
@thekad golang does not guarantee the iteration order of a map.  By using a map here you invalidate the sort above.  You must use a slice.

Reply:
fixed in 4719b2d